src/contracts/atlas/Atlas.sol

```diff
@@ -53,7 +55,9 @@ contract Atlas is Escrow, Factory {
         payable
         returns (bool auctionWon)
     {
-        uint256 _gasMarker = gasleft() + _BASE_TRANSACTION_GAS_USED + (msg.data.length * _CALLDATA_LENGTH_PREMIUM);
+        uint256 _gasMarker = L2_GAS_CALCULATOR == address(0)
+            ? gasleft() + _BASE_TRANSACTION_GAS_USED + (msg.data.length * _CALLDATA_LENGTH_PREMIUM)
+            : IL2GasCalculator(L2_GAS_CALCULATOR).initialGasUsed(msg.data.length);
```

Comment:
In the new case there isn't a `gasleft()` call, so the `initialGasUsed()` call would likely do `gasleft()` in its code. However this can make the code more complicated to reason about, because it'd be a different callframe and `gasleft()` would mean something different (e.g. `gasleft()` in the external call would be capped by the 63/64 rule from EIP150). So it might be better to have it as:

```solidity
gasleft() + IL2GasCalculator(L2_GAS_CALCULATOR).initialGasUsed(msg.data.length)
```

Reply:
This makes sense to me. Will make the changes quick

Reply:
https://github.com/FastLane-Labs/atlas/pull/344/commits/aee1527ee061b911a9805be2a09e53b20b60fd0d